packages/foam-cli/src/commands/janitor.ts

```diff
@@ -0,0 +1,75 @@
+import { Command, flags } from '@oclif/command';
+import * as ora from 'ora';
+import { initializeNoteGraph, generateLinkReferences, generateHeading, getKebabCaseFileName } from 'foam-core';
+import { applyTextEdit } from '../utils/apply-text-edit';
+import { writeFileToDisk } from '../utils/write-file-to-disk';
+import { renameFile } from '../utils/rename-file';
+import * as fs from 'fs'
+
+
+export default class Janitor extends Command {
+  static description = 'Updates link references and heading across all the markdown files in the given workspaces';
+
+  static examples = [
+    `$ foam-cli janitor path-to-foam-workspace
+Successfully generated link references and heading!
+`,
+  ]
+
+  static flags = {
+    help: flags.help({ char: 'h' }),
+  }
+
+  static args = [{ name: 'workspacePath' }]
+
+  async run() {
+    const spinner = ora('Reading Files').start();
+
+    const { args, flags } = this.parse(Janitor)
+
+    const { workspacePath = './' } = args;
+
+    if (fs.existsSync(workspacePath) && fs.lstatSync(workspacePath).isDirectory()) {
+      const graph = await initializeNoteGraph(workspacePath);
+
+      const notes = graph.getNotes();
+
+      spinner.succeed();
+      spinner.text = `${notes.filter(note => note !== undefined).length} files found`;
```

Comment:
Instead of filtering here and then checking again inside notes.map, why not filter once and reuse the same array?

Reply:
Done! Added the code for exiting early in case no files are found.